@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { FormEventHandler } from 'react'
+import { Map } from 'immutable'
+import { withState } from 'reaclette'
+
+import Button from './Button'
+import Input from './Input'
+import IntlMessage from './IntlMessage'
+import Select, { Options } from './Select'
+import { alert } from './Modal'
+
+import XapiConnection, { ObjectsByType, Pif, PifMetrics } from '../libs/xapi'
+
+interface ParentState {
+  objectsByType: ObjectsByType
+  objectsFetched: boolean
+  xapi: XapiConnection
+}
+
+interface State {
+  formRef: React.RefObject<HTMLFormElement>
+  isBonded: boolean
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  _newNetwork: FormEventHandler<HTMLFormElement>
+  _resetForm: () => void
+  _toggleBonded: () => void
+}
+
+interface Computed {
+  pifs?: Map<string, Pif>
+  pifsMetrics?: Map<string, PifMetrics>
+}
+
+const OPTIONS_RENDER_PIF: Options<Pif> = {
+  render: (pif, additionalProps) =>
+    `${pif.device} (${
+      additionalProps?.pifsMetrics?.find((metrics: PifMetrics) => metrics.$ref === pif.metrics)?.device_name ??
+      'unknown'
+    })`,
+  value: pif => pif.$id,
+}
+const OPTIONS_RENDER_BOND_MODE: Options<string[]> = {
+  render: mode => mode,
+  value: mode => mode,
+}
+const BOND_MODE = ['balance-slb', 'active-backup', 'lacp']
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      formRef: React.createRef(),
+      isBonded: false,
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
+      pifsMetrics: state => state.objectsByType.get('PIF_metrics'),
+    },
+    effects: {
+      _newNetwork: async function (e) {
```

Comment:
Why `_` ?

Reply:
`createNetwork`?

Reply:
You can extract this function to `xapi` if it's unrelated to the UI.